src/inspector/value-mirror.cc

```diff
@@ -1434,7 +1434,13 @@ bool ValueMirror::getProperties(v8::Local<v8::Context> context,
   }
 
   auto iterator = v8::debug::PropertyIterator::Create(context, object,
-                                                      nonIndexedPropertiesOnly, params);
+                                                      nonIndexedPropertiesOnly
```

Comment:
As explained [here](https://discord.com/channels/779097926135054346/1212568531611164683/1212675170976669726), I cannot imagine how removing short-circuiting here is not going to have a heavy performance implication?

Reply:
it definitely does have a perf hit associated with it.  I think going through product/protocol improvements later on is likely to make the perf hit less important/noticeable, but definitely worth tabling things until/unless we get those improvements.

I've updated the PR to use a hybrid method where we pass a much larger KeyIterationParams to the PropertyIterator::Create method, then we early out when creating PropertyMirrors.